### PR TITLE
TOD, Travel, and Action AI Legs integration and updates

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/693)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/288)

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
More minimal option:
image
image

**Comment by scott062** [comment]
Maybe hide the legs until loading state for the legs ref data is finished

File: src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx [hunk 1]
```diff
@@ -1,14 +1,16 @@
 import { useCallback, useEffect, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import { shortenString } from 'utilities';
-import { filter, take, takeRight } from 'lodash'; // eslint-disable-line
+import { filter, get, includes, take, takeRight } from 'lodash'; // eslint-disable-line
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx:4]: 🧼?

**Comment by SidelineCory24** [comment]
> **Option 1 - More minimal:** image image
> 
> **Option 2:** image image

I like option 1, but on hover option 2 😅

**Comment by SidelineCory24** [comment]
updating one dropdown is updating them all for that row in the other columns. If it's just on my end disregard

**Comment by SidelineCory24** [comment]
> > updating one dropdown is updating them all for that row in the other columns. If it's just on my end disregard
> 
> Yea, that's to be expected in this PR - i'm going to address it in the AI Legs one; that's when all the logic for individual legs will be set up

back to my corner

File: src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx [hunk 2]
```diff
@@ -224,23 +256,28 @@ const AgendaItemLegs = props => {
...
-                </th>
-                {tr.content}
-              </tr>
-            ))
-          }
-        </tbody>
-      </table>
+      {
+        legsLoading ?
+          <Spinner type="legs" size="small" />
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx:261]: 🚀

**Comment by SidelineCory24** [comment]
@elizabeth-jimenez are we going with option 1 or option 2 for the dropdown border? Other than that everything looks good on my end.

Not sure if Scott had any ideas outside of the spinner, I know he was taking the lead on this review.

File: src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx (hunk 1 above)
Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemLegs/AgendaItemLegs.jsx]: done

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez are we going with option 1 or option 2 for the dropdown border? Other than that everything looks good on my end.
> 
> Not sure if Scott had any ideas outside of the spinner, I know he was taking the lead on this review.

@SidelineCory24 I decided to go with your combo of the two, that was a good suggestion!
🤷‍♀️ idk either lol

**Comment by scott062** [comment]
Yeah sorry I know thats confusing. In the future, if you feel good about it - feel free to approve regardless of who started it. The author can decide if they want to wait for multiple approvals or not. 

However one more thing i noticed. On the hover, it seems to resize by a couple pixels. Idk if thats an easy enough fix or if we just hold off until the next integration PR. But if ya'll can just confirm you see it too.

**Review by SidelineCory24 — APPROVED** [review]
Looks good to go!🚀

**Comment by scott062** [comment]
https://user-images.githubusercontent.com/38047685/178741962-d2a530e5-8671-4bba-aed2-66c7556332cb.mov

**Pull request merged**